pkg/controller/jobs/rayjob/rayjob_webhook.go

```diff
@@ -0,0 +1,131 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rayjob
+
+import (
+	"context"
+
+	rayjobapi "github.com/ray-project/kuberay/ray-operator/apis/ray/v1alpha1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/klog/v2"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
+
+	"sigs.k8s.io/kueue/pkg/controller/jobframework"
+	"sigs.k8s.io/kueue/pkg/util/pointer"
+)
+
+type RayJobWebhook struct {
+	manageJobsWithoutQueueName bool
+}
+
+// SetupWebhook configures the webhook for rayjobapi RayJob.
+func SetupRayJobWebhook(mgr ctrl.Manager, opts ...jobframework.Option) error {
+	options := jobframework.DefaultOptions
+	for _, opt := range opts {
+		opt(&options)
+	}
+	wh := &RayJobWebhook{
+		manageJobsWithoutQueueName: options.ManageJobsWithoutQueueName,
+	}
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(&rayjobapi.RayJob{}).
+		WithDefaulter(wh).
+		WithValidator(wh).
+		Complete()
+}
+
+// +kubebuilder:webhook:path=/mutate-ray-io-v1alpha1-rayjob,mutating=true,failurePolicy=fail,sideEffects=None,groups=ray.io,resources=rayjobs,verbs=create,versions=v1alpha1,name=mrayjob.kb.io,admissionReviewVersions=v1
+
+var _ webhook.CustomDefaulter = &RayJobWebhook{}
+
+// Default implements webhook.CustomDefaulter so a webhook will be registered for the type
+func (w *RayJobWebhook) Default(ctx context.Context, obj runtime.Object) error {
+	job := obj.(*rayjobapi.RayJob)
+	log := ctrl.LoggerFrom(ctx).WithName("rayjob-webhook")
+	log.V(5).Info("Applying defaults", "job", klog.KObj(job))
+	jobframework.ApplyDefaultForSuspend((*RayJob)(job), w.manageJobsWithoutQueueName)
+	return nil
+}
+
+// +kubebuilder:webhook:path=/validate-ray-io-v1alpha1-rayjob,mutating=false,failurePolicy=fail,sideEffects=None,groups=ray.io,resources=rayjobs,verbs=update,versions=v1alpha1,name=vrayjob.kb.io,admissionReviewVersions=v1
+
+var _ webhook.CustomValidator = &RayJobWebhook{}
+
+// ValidateCreate implements webhook.CustomValidator so a webhook will be registered for the type
+func (w *RayJobWebhook) ValidateCreate(ctx context.Context, obj runtime.Object) error {
+	job := obj.(*rayjobapi.RayJob)
+	log := ctrl.LoggerFrom(ctx).WithName("rayjob-webhook")
+	log.Info("Validating create", "job", klog.KObj(job))
+	return w.validateCreate(job).ToAggregate()
+}
+
+func (w *RayJobWebhook) validateCreate(job *rayjobapi.RayJob) field.ErrorList {
+	var allErrors field.ErrorList
+	kueueJob := (*RayJob)(job)
+
+	if w.manageJobsWithoutQueueName || jobframework.QueueName(kueueJob) != "" {
```

Comment:
It is worth the commonizing process to check whether jobs are managed by kueue.

We can work on that as a follow-up PR.

Reply:
Great!